src/main/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImpl.java

```diff
@@ -0,0 +1,149 @@
+package com.pusher.client.channel.impl;
+
+import com.google.gson.Gson;
+import com.pusher.client.AuthorizationFailureException;
+import com.pusher.client.Authorizer;
+import com.pusher.client.channel.PrivateEncryptedChannel;
+import com.pusher.client.channel.PrivateEncryptedChannelEventListener;
+import com.pusher.client.channel.SubscriptionEventListener;
+import com.pusher.client.connection.impl.InternalConnection;
+import com.pusher.client.crypto.nacl.SecretBoxOpener;
+import com.pusher.client.util.Factory;
+
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class PrivateEncryptedChannelImpl extends ChannelImpl implements PrivateEncryptedChannel {
+
+    private static final Gson GSON = new Gson();
+    private final InternalConnection connection;
+    private final Authorizer authorizer;
+    private SecretBoxOpener secretBoxOpener;
+
+    private class PrivateEncryptedChannelData {
+        byte[] auth;
+        final String channelData;
+
+        protected PrivateEncryptedChannelData(String auth, String channelData) {
+            this.auth = auth.getBytes();
+            this.channelData = channelData;
+        }
+
+        public String getAuth() {
+            return new String(auth);
+        }
+
+        public String getChannelData() {
+            return channelData;
+        }
+
+        protected void clearAuthToken(){
+            Arrays.fill(auth, (byte)0);
```

Comment:
Are we okay to use `Arrays.fill` - https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#fill(byte[],%20byte) ?

Reply:
It is OK, why wouldn't it? Just static import and formatting would benefit legibility:
```suggestion
            fill(auth, (byte) 0);
```

Reply:
👍 thanks